wifistats/index.js

```diff
@@ -0,0 +1,129 @@
+const exec          = require('child_process').exec;
+const async         = require('async');
+// The tools
+const airport       = require('./airport');
+const procwireless  = require('./procwireless');
+const netsh         = require('./netsh');
+
+let toolInstance;
+
+/**
+ * Uses all available tools to query for wifi stats, and the one we found to
+ * work we store in global variable {@link toolInstance}.
+ * The result is returned to callback.
+ * We call all tools in parallel and when all finishes we check the results.
+ *
+ * @callback callback delivering results
+ * @param {Error} error if any
+ * @param {object} result stats
+ */
+function initTools(callback) {
+    async.parallel([
+            function (cb) {
+                getStats(
+                    function (err, str) {
+                        cb(null, {err: err, tool: airport, result: str});
+                    }, airport);
+            },
+            function (cb) {
+                getStats(function (err, str) {
+                    cb(null, {err: err, tool: procwireless, result: str});
+                }, procwireless);
+            },
+            function (cb) {
+                getStats(function (err, str) {
+                    cb(null, {err: err, tool: netsh, result: str});
+                }, netsh);
+            }
+        ],
+        function (err, results) {
+            const res = results.find((r) => (r.err == null));
+            if (res) {
+                callback(null, res.tool, res.result);
+            } else {
+                callback(new Error('No known wifi stats tool found'));
+            }
+        }
+    );
+}
+
+/**
+ * Queries for wifi stats using a specific tool or if not specified, uses the
+ * global one which we discover to be working.
+ * @param tool - The tool to use querying for wifi stats.
+ * @callback callback delivering results
+ * @param {Error} error if any
+ * @param {object} result stats
+ */
+function getStats(callback, tool) {
+    if (!tool) {
+        if (toolInstance) {
+            tool = toolInstance;
```

Comment:
Why did you want getStats to set tool? It would surprise me that a function called getStats is setting tool.

Reply:
This is the initialization on the first call, which founds which tool will be used, checking all operating systems.